hw/mcu/nordic/nrf52xxx/src/nrf52_clock.c

```diff
@@ -39,7 +39,12 @@ nrf52_clock_hfxo_request(void)
     __HAL_DISABLE_INTERRUPTS(ctx);
     assert(nrf52_clock_hfxo_refcnt < 0xff);
     if (nrf52_clock_hfxo_refcnt == 0) {
+        /* Make sure that the HFXO has not previously been started manually */
+        assert((NRF_CLOCK->HFCLKSTAT & CLOCK_HFCLKSTAT_STATE_Msk) !=
+            (CLOCK_HFCLKSTAT_STATE_Running << CLOCK_HFCLKSTAT_STATE_Pos));
```

Comment:
When I compile a blinky and try to run it on an nrf52840 board it breaks on this assert.
I just looked at some [nrfx code](https://github.com/NordicSemiconductor/nrfx/blob/master/hal/nrf_clock.h#L505) which verifies whether hfclk is running and I found a few differences. @amrbekhit Can you change and verify this please?

Reply:
also I tried this out and it worked https://github.com/mlaz/mynewt-core/pull/new/nrf52_lp_timer